src/query/service/src/pipelines/processors/transforms/hash_join/hash_join_build_state.rs

```diff
@@ -229,15 +264,30 @@ impl HashJoinBuildState {
                     .build_num_rows
             };
 
-            let build_chunks =
-                &mut unsafe { &mut *self.hash_join_state.build_state.get() }.build_chunks;
-            if build_num_rows <= INLIST_RUNTIME_FILTER_THRESHOLD {
-                *build_chunks = unsafe {
-                    (*self.hash_join_state.build_state.get())
-                        .generation_state
-                        .chunks
-                        .clone()
-                };
+            let build_chunks = unsafe {
+                (*self.hash_join_state.build_state.get())
+                    .generation_state
+                    .chunks
+                    .clone()
+            };
+
```

Comment:
if it takes place spill, can we hold these all entirely?

Reply:
Spill and runtime filter are in conflict. We won't enable runtime filter when spilling.